gui/src/avopenglwidget.cpp

```diff
@@ -93,6 +141,15 @@ AVOpenGLWidget::AVOpenGLWidget(VideoDecoder *decoder, QWidget *parent)
 	: QOpenGLWidget(parent),
 	decoder(decoder)
 {
+	for(auto &cc: conversion_configs)
+	{
+		if(decoder->PixelFormat() == cc.pixel_format)
+		{
+			conversion_config = &cc;
+			break;
+		}
+	}
+
```

Comment:
I don't know how to return an error here if PixelFormat() is not found in available conversion configs. Just throw?

Reply:
Yes, you can throw this: https://github.com/thestr4ng3r/chiaki/blob/master/gui/include/exception.h